test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_test.clj

```diff
@@ -744,6 +748,26 @@
              (is (= 200 (:status @response)))
              (is (= hello-body (:body @response))))))))))
 
+(deftest double-stop-test
+  (testing "if the stop lifecycle is called more than once, we handle that gracefully and quietly"
+    (let [log-messages (atom [])]
+      (tk-log-testutils/with-logging-to-atom tk-log/root-logger-name log-messages
+        (let [app (tk-bootstrap/bootstrap-services-with-config
+                   [jetty9-service]
+                   {:webserver {:port 8080}})]
+          (tk-app/stop app)
```

Comment:
I understand that the `TrapperkeeperApp` will swalow the exception thrown by the second `stop` call and so looking at the log file might be the only way to determine the benefit of these code changes at more of a "system level".  In this case, though, I'd probably just have the test get the `jetty9-service` instance, call `stop` on it directly and validate that no exception is thrown.  Not as standard of a thing to do from a Trapperkeeper app lifecycle perspective but it seems like it would be more targeted around the specific problem we're trying to validate has been fixed.